src/OddJobs/Job.hs

```diff
@@ -503,12 +508,17 @@ jobPoller = do
   lockTimeout <- getDefaultJobTimeout
   log LevelInfo $ LogText $ toS $ "Starting the job monitor via DB polling with processName=" <> processName
   concurrencyControlFn <- getConcurrencyControlFn
-  withResource pool $ \pollerDbConn -> forever $ concurrencyControlFn >>= \case
-    False -> do
-      log LevelWarn $ LogText $ "NOT polling the job queue due to concurrency control"
-      -- If we can't run any jobs ATM, relax and wait for resources to free up
-      delayAction
-    True -> do
+  withResource pool $ \pollerDbConn -> forever $
+    concurrencyControlFn >>= \case
+      False -> do
+        log LevelWarn $ LogText $ "NOT polling the job queue due to concurrency control"
+        -- If we can't run any jobs ATM, relax and wait for resources to free up
+        delayAction
+      True -> polJob processName pollerDbConn tname lockTimeout
+
+-- | polls a single job and runs it.
+polJob :: (HasJobRunner m) => String -> Connection -> TableName -> Seconds -> m ()
+polJob processName pollerDbConn tname lockTimeout = do
```

Comment:
my expectation was this would also update the status if the polled task fails, but I don't think it does that right now.

I pulled it away from the outer function so I don't have to deal with all concurency complexity.